pkg/proxy/winuserspace/proxier.go

```diff
@@ -194,105 +178,139 @@ func (proxier *Proxier) SyncLoop() {
 	}
 }
 
-// Ensure that portals exist for all services.
-func (proxier *Proxier) ensurePortals() {
-	proxier.mu.Lock()
-	defer proxier.mu.Unlock()
-	// NB: This does not remove rules that should not be present.
-	for name, info := range proxier.serviceMap {
-		err := proxier.openPortal(name, info)
-		if err != nil {
-			glog.Errorf("Failed to ensure portal for %q: %v", name, err)
-		}
-	}
-}
-
 // cleanupStaleStickySessions cleans up any stale sticky session records in the hash map.
 func (proxier *Proxier) cleanupStaleStickySessions() {
 	proxier.mu.Lock()
 	defer proxier.mu.Unlock()
+	servicePortNameMap := make(map[proxy.ServicePortName]bool)
 	for name := range proxier.serviceMap {
+		servicePortName := proxy.ServicePortName{NamespacedName: types.NamespacedName{Namespace: name.Namespace, Name: name.Name}, Port: name.Port}
+		servicePortNameMap[servicePortName] = true
+	}
+	for name := range servicePortNameMap {
```

Comment:
this seems weird to have this second loop. Why not just do the cleanup above instead of adding `true` to the map?